### PR TITLE
CLI + MSP: Disable failsafe on throttle cut

### PR DESCRIPTION
**Pull request by gaelj**
new CLI command + extra byte in MSP_FAILSAFE_CONFIG : `disable_failsafe_on_throttle_cut`
addresses https://github.com/iNavFlight/inav/issues/339

### CODE REVIEW

**Comment by gaelj** [comment]
Note: the MSP change does not break the current configurator.

**Comment by digitalentity** [comment]
Is this really needed? Failsafe is supposed to kick in when RX is lost - therefore throttle is also being overriden.

**Comment by DaleKramer** [comment]
@digitalentity Because I can be flying on a gas motor in airplane mode (with throttle cut (1025) to my FC) and with a separate throttle servo, RX (with many sats) and servos, there is a good possibility that I fly out of range of my single sat FC. At this point I can not afford to have an FC failsafe initiated during throttle cut (it will destabilize the vehicle).

**Comment by digitalentity** [comment]
@DaleKramer I understand your concern. However, out of range condition will also cause `rxfail` mechanism to override throttle channen to low value which, in turn, will disable stage 2 failsafe if your idea is implemented.

Why would you need failsafe at all?

**Comment by DaleKramer** [comment]
Because, when I am in hover mode, there is a still a chance of being out of range of the single sat FC and I still want FC failsafe then.  Hover mode is activated when I do NOT have my throttle cut switch active (TX sends out 1100 when throttle stick is idle with throttle cut switch NOT active) after I transition from airplane mode to hover mode.

**Comment by digitalentity** [comment]
Ok, I see, you want whole failsafe system to be active only when last valid throttle received from RX was above min_check. Sounds valid.

File: src/main/mw.c [hunk 1]
```diff
@@ -280,7 +280,7 @@ void processRx(void)
 
     updateRSSI(currentTime);
 
-    if (feature(FEATURE_FAILSAFE)) {
+    if (feature(FEATURE_FAILSAFE) && (rcData[THROTTLE] > 1050 || !(masterConfig.failsafeConfig.disable_failsafe_on_throttle_cut))) {
```
Comment (digitalentity) [inline src/main/mw.c]: this does not accound for `rxfail` mechanism (stage 1 of failsafe). When signal is lost `rxfail 3 a` will be active and throttle will be driven low, effectuvely failsafe will be never activated.

Comment (digitalentity) [inline src/main/mw.c]: I'd like to avoid hardcoded values. I think we should use `min_check` here.

**Comment by digitalentity** [comment]
For this logic to work correctly we need to modify `detectAndApplySignalLossBehaviour()` to keep track of last valid received channel values so we can use them to make decisions.

**Comment by DaleKramer** [comment]
Yes, also stopping Stage 1 would be better.  The hard coded value was because I was assuming that I would be having to do this in my personal branch.  

Thank-you for helping with this, I was afraid my kludge may have missed something :)

**Comment by oleost** [comment]
This solution I like.
Also this would be handy with auto arm branch for fixed wing #375

**Comment by DaleKramer** [comment]
I am in a position to test this now but being a git novice I am unsure of how to add these changes to my master.  I have TortoiseGIT and GIT GUI.  

Can someone help me with a GIT procedure, otherwise I will have to paste these changes into my files?

FOR FUTURE REFERENCE THIS IS HOW I BROUGHT IN THIS PULL REQUEST WITH TortoiseGIT:

pullrequest

**Comment by DaleKramer** [comment]
I can now confirm that this code works great and as expected, thanks!

I am not sure what the conflicts are that are mentioned below.

**Comment by gaelj** [comment]
Well, this is strange. I rebased against latest master and didn't get any conflicts !
Maybe some commit to master conflicted at some point, and then that particular change got reverted or changed again in a way that is compatible with this PR.
Anyway I force pushed the rebase so it is ready for merging if you're happy with it, @digitalentity.

**Comment by digitalentity** [comment]
I'm going to merge this soon. @gaelj can you do an according change to configurator?

**Comment by gaelj** [comment]
Will do. It'd be nice to merge this before the configurator change so as to avoid conflicts & rebase :)

**Comment by digitalentity** [comment]
@gaelj can you please rebase?

**Comment by oleost** [comment]
#1475

**Pull request closed without merging**